boards/arm/lpcxpresso55s28/lpcxpresso55s28_common.dtsi

```diff
@@ -0,0 +1,179 @@
+		spi-8 = &hs_lspi;
+		usart-0 = &flexcomm0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		green_led: led_1 {
+			gpios = <&gpio1 7 0>;
+			label = "User LD2";
+			status = "disabled";
+		};
+		blue_led: led_2 {
+			gpios = <&gpio1 4 0>;
+			label = "User LD3";
+			status = "disabled";
+		};
+		red_led: led_3 {
+			gpios = <&gpio1 6 0>;
+			label = "User LD4";
+			status = "disabled";
+		};
+	};
+
+	mikrobus_header: mikrobus-connector {
+		compatible = "mikro-bus";
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map =	<0 0 &gpio0 16 0>,	/* AN  */
+				/* Not a GPIO*/		/* RST */
+				<2 0 &gpio1 1 0>,	/* CS   */
+				<3 0 &gpio1 2 0>,	/* SCK  */
+				<4 0 &gpio1 3 0>,	/* MISO */
+				<5 0 &gpio0 26 0>,	/* MOSI */
+							/* +3.3V */
+							/* GND */
+				<6 0 &gpio1 15 0>,	/* PWM  */
+				<7 0 &gpio1 18 0>,	/* INT  */
+				<8 0 &gpio0 27 0>,	/* RX   */
+				<9 0 &gpio1 24 0>,	/* TX   */
+				<10 0 &gpio1 20 0>,	/* SCL  */
+				<11 0 &gpio1 21 0>;	/* SDA  */
+							/* +5V */
+							/* GND */
+	};
+
+	arduino_header: arduino-connector {
+		compatible = "arduino-header-r3";
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map =	<0 0 &gpio0 15 0>,	/* A0 */
+				<1 0 &gpio0 23 0>,	/* A1 */
+				<2 0 &gpio0 0 0>,	/* A2 */
+				<3 0 &gpio1 31 0>,	/* A3 */
+				<4 0 &gpio0 13 0>,	/* A4 */
+				<5 0 &gpio0 14 0>,	/* A5 */
+				<6 0 &gpio1 24 0>,	/* D0 */
+				<7 0 &gpio0 27 0>,	/* D1 */
+				<8 0 &gpio0 15 0>,	/* D2 */
+				<9 0 &gpio1 15 0>,	/* D3 */
+				<10 0 &gpio1 17 0>,	/* D4 */
+				<11 0 &gpio1 14 0>,	/* D5 */
+				<12 0 &gpio1 10 0>,	/* D6 */
+				<13 0 &gpio1 9 0>,	/* D7 */
+				<14 0 &gpio1 5 0>,	/* D8 */
+				<15 0 &gpio1 8 0>,	/* D9 */
+				<16 0 &gpio1 1 0>,	/* D10 */
+				<17 0 &gpio0 26 0>,	/* D11 */
+				<18 0 &gpio1 3 0>,	/* D12 */
+				<19 0 &gpio1 2 0>,	/* D13 */
+				<20 0 &gpio1 21 0>,	/* D14 */
+				<21 0 &gpio1 20 0>;	/* D15 */
+	};
+};
+
+&flexcomm0 {
+	compatible = "nxp,lpc-usart";
+	current-speed = <115200>;
+};
+
+&flash0 {
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		/* Please enable mcuboot's swap move, no scratch */
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x00020000>;
+			read-only;
+		};
+		slot0_partition: partition@20000 {
+			label = "image-0";
+			reg = <0x00020000 0x00020000>;
+		};
+		slot1_partition: partition@40000 {
+			label = "image-1";
+			reg = <0x00040000 0x00018000>;
+		};
+		slot2_partition: partition@58000 {
+			label = "image-2";
+			reg = <0x00058000 0x00020000>;
+		};
+		slot3_partition: partition@78000 {
+			label = "image-3";
+			reg = <0x00078000 0x00018000>;
+		};
+		/*
+		 * The flash starting at 0x90000 and ending at
+		 * 0x9D7FF is reserved for the application.
+		 */
+		storage_partition: partition@90000 {
+			label = "storage";
+			reg = <0x00090000 0x0000d800>;
```

Comment:
Since this device only has 512kB flash and not 640kB, how should i partition it here?

Reply:
flash partitioning is now completely removed